pkg/search/postgres/query/common.go

```diff
@@ -46,7 +49,17 @@ func (f *SelectQueryField) PathForSelectPortion() string {
 	if f.Alias == "" {
 		return f.SelectPath
 	}
-	return fmt.Sprintf("%s as %s", f.SelectPath, f.Alias)
+	return fmt.Sprintf("%s %s", f.SelectPath, f.Alias)
+}
+
+// MustAggregate returns true if the select field needs to aggregated.
+func (f *SelectQueryField) MustAggregate() bool {
+	return !f.FromGroupBy && !f.DerivedField
```

Comment:
What is the difference between this func and the AggrAndUnnest? What is the decision point to unnest or not? When is this needed? The check condition is the same in both 🤔

Reply:
ironed out the code. Removed unused.